framework/src/base/Moose.C

```diff
@@ -358,6 +359,7 @@ registerExecFlags(Factory & factory)
   registerExecFlag(EXEC_SUBDOMAIN);
   registerExecFlag(EXEC_PRE_DISPLACE);
   registerExecFlag(EXEC_SAME_AS_MULTIAPP);
+  registerExecFlag(EXEC_PRE_INITIAL);
```

Comment:
This seems like a bad idea. Am I correct that any system could potentially attempt to execute on "pre_initial" now? We've spent quite a bit of time making sure that we have called `initialSetup()` before we attempt to call the first execute on any MooseObject. This new flag completely by-passes that now. 

Honestly, I'd love to be able to split up the initialization and execution for initial but we've tried and there is no way to do it. There are too many interdependencies to go that direction.